src/typescript/azure-openapi-validator/rules/PathParameterMustNotBeEmpty.ts

```diff
@@ -0,0 +1,22 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from '../rule';
+export const PathParametersMustNotBeEmpty: string = "PathParametersMustNotBeEmpty";
+
+rules.push({
+  id: "R2067",
+  name: PathParametersMustNotBeEmpty,
+  severity: "warning",
+  category: "RPCViolation",
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.arm,
+
+  appliesTo_JsonQuery: "$.paths..parameters[?(@.in=='path')]",
+  run: function* (doc, node, path) {
+    if ((typeof node.minLength !== "number") || (node.minLength < 1)) {
+      yield { message: `Path parameters should should have the 'minLength' attribute with a value > 0: Parameter '${node.name}' in:'${path[1]}'. `, location: path };
```

Comment:
Since this is a warning and we're not sure if the service may for some reason want to deal with empty strings, we may want to update the message to something like: "If your service requires non-empty path parameters, please add 'minLength' ... ".
You should have a corresponding PR to update documentation at https://github.com/Azure/azure-rest-api-specs/blob/current/documentation/openapi-authoring-automated-guidelines.md (that you can get reviewed and would merge once this one is merged) - You can explain on the "why" that including this property helps the SDK check for this instead of making a call to the service with an empty string.